File: mobile/src/main/java/uk/co/appsbystudio/geoshare/Application.java
package uk.co.appsbystudio.geoshare;

import android.content.Context;

public class Application extends android.app.Application {

    private static Application instance;
    private static Context context;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;

        this.setContext(getApplicationContext());
    }

    public static Application getInstance() {
        return instance;
    }

    public static Context getAppContext() {
        return context;
    }

    public void setContext(Context context) {
        this.context = context;
    }
}

File: mobile/src/main/java/uk/co/appsbystudio/geoshare/Application.kt
package uk.co.appsbystudio.geoshare

import android.content.Context
import com.google.firebase.FirebaseApp
import com.google.firebase.database.FirebaseDatabase

class Application : android.app.Application() {
    init {
        instance = this
    }

    companion object {

        private var instance: Application? = null

        val context: Context?
            get() = instance
    }

    override fun onCreate() {
        super.onCreate()
        if (!FirebaseApp.getApps(this).isEmpty()) {
            FirebaseDatabase.getInstance().setPersistenceEnabled(true)
        }
    }
}
